lib/plugins/static.js

```diff
@@ -55,29 +56,48 @@ function serveStatic(options) {
             res.handledGzip();
         }
 
-        var fstream = fs.createReadStream(file + (isGzip ? '.gz' : ''));
         var maxAge = opts.maxAge === undefined ? 3600 : opts.maxAge;
-        fstream.once('open', function (fd) {
-            res.cache({maxAge: maxAge});
-            res.set('Content-Length', stats.size);
-            res.set('Content-Type', mime.lookup(file));
-            res.set('Last-Modified', stats.mtime);
-
-            if (opts.charSet) {
-                var type = res.getHeader('Content-Type') +
-                    '; charset=' + opts.charSet;
-                res.setHeader('Content-Type', type);
-            }
+        res.cache({maxAge: maxAge});
+        res.set('Content-Length', stats.size);
+        res.set('Content-Type', mime.lookup(file));
+        res.set('Last-Modified', stats.mtime);
+
+        if (opts.charSet) {
+            var type = res.getHeader('Content-Type') +
+                '; charset=' + opts.charSet;
+            res.setHeader('Content-Type', type);
```

Comment:
I don't know: I only moved that code, I didn't write it. It looks like it appends it to the current content-type.